backend/src/sync/client.rs

```diff
@@ -287,6 +288,58 @@ impl OcClient {
     }
 
 
+    pub async fn create_playlist(
+        &self,
+        title: &str,
+        description: Option<&str>,
+        creator: &str,
+        entries: &[String],
+        acl: &[AclInputEntry],
+    ) -> Result<CreatePlaylistResponse> {
+        let access_policy: Vec<AclInput> = acl.iter()
+            .flat_map(|entry| entry.actions.iter()
+                .map(|action| AclInput {
+                    allow: true,
+                    action: action.clone(),
+                    role: entry.role.clone(),
+                }))
+            .collect();
+
+        let entries: Vec<serde_json::Value> = entries.into_iter().map(|e| {
+            serde_json::json!({
+                "contentId": e,
+                "type": "EVENT",
+            })
+        }).collect();
+
+        let playlist = serde_json::json!({
+            "title": title,
+            "description": description,
+            "creator": creator,
+            "entries": entries,
+            "accessControlEntries": access_policy,
+        });
+
+        let params = Serializer::new(String::new())
+            .append_pair("playlist", &serde_json::to_string(&playlist).expect("Failed to serialize"))
+            .finish();
+
+        let req = self.authed_req_builder(&self.external_api_node, "/api/playlists/")
+            .method(http::Method::POST)
+            .header(http::header::CONTENT_TYPE, "application/x-www-form-urlencoded")
+            .body(params.into())
+            .expect("failed to build request");
+
+        let uri = req.uri().clone();
+        let response = self.http_client.request(req).await
+            .with_context(|| format!("HTTP request failed (uri: '{uri}'"))?;
```

Comment:
There is also lots of duplication between this and other methods. Maybe see if you can extract common logic into "make x-www-form-urlencoded request" or something?

Reply:
I took the liberty and refactored some more stuff in this file.